kernel/threads.h

```diff
@@ -22,6 +23,12 @@ enum fpu_state {
   THREAD_FPU_STATE_ACTIVE,    /* there's an FPU buf allocated */
 };
 
+enum ipc_state {
+  IPC_NOT_RECEIVING,
+  IPC_RECEIVING,      /* currently waiting for an ipc message */
+  IPC_RECEIVED,       /* in the middle of receiving an ipc handoff */
```

Comment:
This state is a little sad.  With the current structure, where you drop down into 'schedule' before reaching the other thread, it's probably the best you can do, though.